src/river/river.py

```
@@ -0,0 +1,337 @@
+
+        sx, sy, start_z = feat["start_x"], feat["start_y"], feat["start_z"]
+        ex, ey, end_z = feat["end_x"], feat["end_y"], feat["end_z"]
+
+        # Проверка начальной точки
+        if (
+            sx is not None
+            and sy is not None
+            and start_z is not None
+            and start_z == max_z
+        ):
+            if (sx, sy) not in end_points:
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)